### PR TITLE
OSSMDOC-474: Publish service mesh docs to ROSA

### PR DESCRIPTION
**Pull request by rh-tokeefe**
Version(s):

Issue: https://issues.redhat.com/browse/OSSMDOC-474

Link to ROSA docs preview: http://file.bos.redhat.com/tokeefe/OSSMDOC-474-New/welcome/index.html

NOTE:
Automatic preview functionality is currently not available.
  * OpenShift documentation team members (core and aligned) must include a link to a locally generated preview for PRs that update the rendered build in any way.
  * External contributors can request a generated preview from the OpenShift documentation team. --->

Additional information:

### CODE REVIEW

**Comment by skondkar** [comment]
LGTM

**Comment by rh-tokeefe** [comment]
@okashi18 please review. Thanks!

File: service_mesh/v1x/installing-ossm.adoc [hunk 1]
```diff
@@ -54,7 +54,12 @@ include::modules/ossm-member-roll-modify.adoc[leveloffset=+1]
 If you choose to update manually, the Operator Lifecycle Manager (OLM) controls the installation, upgrade, and role-based access control (RBAC) of Operators in a cluster. OLM runs by default in {product-title}.
 OLM uses CatalogSources, which use the Operator Registry API, to query for available Operators as well as upgrades for installed Operators.
 
+ifdef::openshift-enterprise[]
 * For more information about how {product-title} handled upgrades, refer to the xref:../../operators/understanding/olm/olm-understanding-olm.adoc#olm-overview_olm-understanding-olm[Operator Lifecycle Manager] documentation.
+endif::[]
+
+ifdef::openshift-rosa[]
```
Comment (EricPonvelle) [inline service_mesh/v1x/installing-ossm.adoc:61]: Omit

File: modules/ossm-routing-ingress.adoc [hunk 2]
```diff
@@ -23,7 +23,13 @@ That command returns the `NAME`, `TYPE`, `CLUSTER-IP`, `EXTERNAL-IP`,    `PORT(S
 
 If the `EXTERNAL-IP` value is set, your environment has an external load balancer that you can use for the ingress gateway.
 
-If the `EXTERNAL-IP` value is `<none>`, or perpetually `<pending>`, your environment does not provide an external load balancer for the ingress gateway. You can access the gateway using the service's xref:../../networking/configuring-node-port-service-range.adoc[node port].
+If the `EXTERNAL-IP` value is `<none>`, or perpetually `<pending>`, your environment does not provide an external load balancer for the ingress gateway.
+ifdef::openshift-enterprise[]
+You can access the gateway using the service's xref:../../networking/configuring-node-port-service-range.adoc[node port].
+endif::[]
+
+ifdef::openshift-rosa[]
```
Comment (EricPonvelle) [inline modules/ossm-routing-ingress.adoc:31]: Omit

File: modules/ossm-config-sampling.adoc [hunk 3]
```diff
@@ -21,7 +21,13 @@ In a basic installation, `spec.tracing.sampling` is set to `10000`, which sample
...
 
-The Jaeger remote sampling rate applies to applications that are external to the Service Mesh, and do not use the Envoy proxy, such as a database. This sampling rate determines how much data the distributed tracing system collects and stores. For more information, see xref:../../distr_tracing/distr_tracing_install/distr-tracing-deploying-jaeger.adoc#distr-tracing-config-sampling_deploying-distributed-tracing-platform[Distributed tracing configuration options].
+The Jaeger remote sampling rate applies to applications that are external to the Service Mesh, and do not use the Envoy proxy, such as a database. This sampling rate determines how much data the distributed tracing system collects and stores.
+ifdef::openshift-enterprise[]
+For more information, see xref:../../distr_tracing/distr_tracing_install/distr-tracing-deploying-jaeger.adoc#distr-tracing-config-sampling_deploying-distributed-tracing-platform[Distributed tracing configuration options].
+endif::[]
+
+ifdef::openshift-rosa[]
```
Comment (EricPonvelle) [inline modules/ossm-config-sampling.adoc:29]: Omit

File: modules/ossm-auto-route-annotations.adoc [hunk 4]
```diff
@@ -5,6 +5,13 @@
 [id="ossm-auto-route-annotations_{context}"]
 = {SMProductName} route annotations
 
-Sometimes specific annotations are needed in an OpenShift Route. For example, some advanced features in OpenShift Routes are managed via xref:../../networking/routes/route-configuration.adoc[special annotations]. For this and other use cases, {SMProductName} will copy all annotations present in the Istio Gateway resource (with the exception of those starting with `kubectl.kubernetes.io`) into the managed OpenShift Route resource.
+Sometimes specific annotations are needed in an OpenShift Route.
+ifdef::openshift-enterprise[]
+For example, some advanced features in OpenShift Routes are managed via xref:../../networking/routes/route-configuration.adoc[special annotations].
+endif::[]
+
+ifdef::openshift-rosa[]
```
Comment (EricPonvelle) [inline modules/ossm-auto-route-annotations.adoc:13]: Omit

File: modules/ossm-configuring-the-threescale-wasm-auth-module.adoc [hunk 5]
```diff
@@ -10,7 +10,15 @@ Cluster administrators on {product-title} can configure the `threescale-wasm-aut
...
 
-{SMProductShortName} provides a xref:../../operators/understanding/crds/crd-extending-api-with-crds.adoc#crd-extending-api-with-crds[custom resource definition] to specify and apply Proxy-WASM extensions to sidecar proxies, known as xref:../../service_mesh/v2x/ossm-extensions.adoc#ossm-extensions[`ServiceMeshExtension`]. {SMProductShortName} applies this custom resource to the set of workloads that require HTTP API management with 3scale.
+
+{SMProductShortName} provides a custom resource definition to specify and apply Proxy-WASM extensions to sidecar proxies, known as xref:../../service_mesh/v2x/ossm-extensions.adoc#ossm-extensions[`ServiceMeshExtension`].  {SMProductShortName} applies this custom resource to the set of workloads that require HTTP API management with 3scale.
+
+ifdef::openshift-enterprise[]
+See xref:../../operators/understanding/crds/crd-extending-api-with-crds.adoc#crd-extending-api-with-crds[custom resource definition] for more information.
+endif::[]
+
+ifdef::openshift-rosa[]
```
Comment (EricPonvelle) [inline modules/ossm-configuring-the-threescale-wasm-auth-module.adoc:20]: Omit

File: service_mesh/v1x/ossm-traffic-manage.adoc [hunk 6]
```diff
@@ -22,6 +22,11 @@ include::modules/ossm-routing-ingress.adoc[leveloffset=+1]
 
 include::modules/ossm-auto-route-1x.adoc[leveloffset=+1]
 
+ifdef::openshift-enterprise[]
 == Links
 
 For more information about configuring an {product-title} wildcard policy, see xref:../../networking/ingress-operator.adoc#using-wildcard-routes_configuring-ingress[Using wildcard routes].
+endif::[]
+
+ifdef::openshift-rosa[]
```
Comment (EricPonvelle) [inline service_mesh/v1x/ossm-traffic-manage.adoc:31]: Omit

File: service_mesh/v1x/ossm-custom-resources.adoc [hunk 7]
```diff
@@ -39,6 +44,11 @@ include::modules/ossm-jaeger-config-elasticsearch-v1x.adoc[leveloffset=+2]
 
 include::modules/ossm-jaeger-config-es-cleaner-v1x.adoc[leveloffset=+2]
 
+ifdef::openshift-enterprise[]
 For more information about configuring Elasticsearch with {product-title}, see  xref:../../logging/config/cluster-logging-log-store.adoc[Configuring the log store].
+endif::[]
+
+ifdef::openshift-rosa[]
```
Comment (EricPonvelle) [inline service_mesh/v1x/ossm-custom-resources.adoc:51]: Omit

**Review by EricPonvelle — COMMENTED** [review]
I recommend omitting the empty conditionals for ROSA.

**Comment by rh-tokeefe** [comment]
@EricPonvelle I've completed the updates. Thank you for your review!

**Comment by johnwilkins** [comment]
LGTM. The changes look like previously published modules with new conditionals when it applies to openshift-enterprise along with new topic map entries for ROSA, and the topic map and conditionals appear to be working just fine.

**Pull request merged**